API.md

```diff
@@ -1,269 +0,0 @@
-API
```

Comment:
I think we should keep the documentation, because it is way easier for someone looking at this particular API for the first time to figure out what's going on based on the documentation, rather than having to interpret the code. Its not a matter of them not knowing how to read the code, but they shouldn't have to do it because the documentation exists precisely for that,